src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -874,6 +874,14 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                 AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeNullableContextAttribute(this, nullableContextValue));
             }
 
+            bool isCovariantOverride = this.IsOverride && !this.OverriddenMethod.ReturnType.Equals(this.ReturnType, TypeCompareKind.AllIgnoreOptions);
+            if (isCovariantOverride)
+            {
+                // If present, we add PreserveBaseOverridesAttribute to covariant overrides.
```

Comment:
>If present, [](start = 19, length = 11)

Why if present? Isn't this attribute required to enforce proper behavior at runtime?

Reply:
Yes, indeed it is. How to deal with this is tracked by [https://github.com/dotnet/roslyn/issues/44208](https://github.com/dotnet/roslyn/issues/44208)

---
In reply to: [441984151](https://github.com/dotnet/roslyn/pull/44025#discussion_r441984151) [](ancestors = 441984151)